### DISCUSSION TITLE
Github Actions stuck at "Queued" for more than 24 hours

### DISCUSSION THREAD

**Discussion by doorgan** [community/community#56360] [Actions] [answered]
### Select Topic Area

Bug

### Body

The Actions for my repository https://github.com/doorgan/sourceror are stuck at "Queued" and they don't seem to be going to run any time soon: https://github.com/doorgan/sourceror/actions/runs/5060935883/jobs/9114456314

There haven't been any changes to the Actions configuration, and there hasn't been an Actions run in more than a month(last one failed due to a GH issue).

Is there anything I can do to fix it?

Thank you

**Comment by airtower-luna** [comment, marked as answer]
That there hasn't been any change is exactly the problem: Your jobs used `ubuntu-18.04` runners. Those have been deprecated for a while and were removed in April, and now your jobs queue endlessly because they simply can't get a runner. You'll have to update the jobs to use currently supported runners (best switch right to `ubuntu-22.04`, if possible).

See: https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/

**Reply by doorgan** [reply]
That seems to be it, thanks!

**Comment by RobSmyth** [comment]
I came across this issue and I'm very surprised that runners get depreciated as it impacts reproducibility. So FYI ...

If runners (environments) are depreciated from time to time that means that it may not be possible to reproduce an old build. I take the two fundamental tasks for a build system are:

- Reproducibility
- Traceability

If it is not possible to rebuild using the same environment, then reproducibility is gone.

Another note ... if the build system does not provide a build number (run and attempt numbers are not build numbers) then traceability is gone too.

**Reply by jannes-io** [reply]
There are many runners on a single host machine. I do not want to run my GH actions on a machine that is also running actions for deprecated/unsupported versions of operating systems that may have critical vulnerabilities.

GH has a responsibility to its customers to provide a secure way to run their actions, many actions contain highly sensitive data.

I don't know if it's possible to run old runner images on self-hosted runners.. But that would be the only place where GH should allow it...

**Comment by RobSmyth** [comment]
A second thought ... I think this is a bug as:

- The message 'queued' is misleading. The services required do not exist, it ought to fail. Message is inadequate.
- It means that the only runner that will run long term is "latest". Any other choice will eventually fail.
